packages/jbrowse-plugin-apollo/src/session.ts

```diff
@@ -126,6 +237,23 @@ const ClientDataStore = types
     deleteAssembly(assemblyId: string) {
       self.assemblies.delete(assemblyId)
     },
+    getLocations() {
+      // console.log(`0 VIEWS: ${JSON.stringify(self)}`)
+      console.log(
+        `1 VIEWS: ${JSON.stringify(self as unknown as AbstractSessionModel)}`,
+      )
+      // console.log(`2 VIEWS: ${JSON.stringify((self as unknown as AbstractSessionModel).views)}`)
+      // const locations = []
+      // for (const view of (self as unknown as AbstractSessionModel).views) {
+      //   if (view.type === 'LinearGenomeView') {
+      //     console.log(`VIEW: ${JSON.stringify(view)}`)
+      //     // const {dynamicBlocks} = view
+      //     // // view and get location
+      //     // console.log(`BLOCKS: ${JSON.stringify(dynamicBlocks)}`)
+      //     // // run in https://developer.mozilla.org/en-US/docs/Web/API/setInterval
+      //   }
+      // }
+    },
```

Comment:
This needs to be on the session, not the data store (e.g. above `apolloSetSelectedFeature`). This should work to get you the locations:

```ts
getLocations() {
  const locations = []
  for (const view of self.views) {
    if (view.type === 'LinearGenomeView') {
      const { dynamicBlocks } = view as LinearGenomeViewModel
      dynamicBlocks.forEach((block) => {
        if (block.regionNumber !== undefined) {
          const { assemblyName, refName, start, end } = block
          locations.push({ assemblyName, refName, start, end })
        }
      })
    }
  }
  return locations
}
```

Reply:
Fixed